### PR TITLE
143 add and edit meetings on the calendar

### PR DESCRIPTION
**Pull request by DanR2003**
This PR introduces two new screens: EditMeeting and AddMeeting.

- EditMeeting: Accessible by selecting a meeting in the calendar, this screen allows users to update the meeting's location or date, or to delete the meeting.

- AddMeeting: Available via the calendar icon in the chat bar, this screen enables users to schedule a meeting with the friend they are currently chatting with.

Additionally:

- Only meetings involving the user (either as creatorId or friendId) are now displayed in the calendar.

- Documentation has been added, along with unit and UI tests for the meeting screens, MeetingViewModel, and MeetingRepositoryFirestore.

- Unused, commented code was removed for clarity.

### CODE REVIEW

**Review by marinovdeyan — CHANGES_REQUESTED** [review]
Very good addition to the app. The functionality is working and well implemented. I'm glad you implemented my suggestion from your previous PR in order to better filter meetings.

To be changed before merging :
- Remove commented code in MeetingRepositoryFirestore.kt
- Fix poorly named variable in AddMeeting.kt

Suggestions for future PRs around this feature:
- Add the name of the person with whom you're meeting in the AddMeeting and EditMeeting screens. Right now you have to remember what was written on the Calendar screen before you click to edit or which friend you are currently talking to before creating an event.
- Adding more test for the MeetingViewModel in order to increase line coverage and cover more cases. Some of the functions are still untested for now. Having tested the functionality on my phone, I can see that it is currently working, so I am not requesting this as a mandatory change for now, as I understand that more tests are very hard to implement due to the complexity of the mocking.

Good luck with the corrections!

File: app/build.gradle.kts
Comment (ifaresse) [inline app/build.gradle.kts:1]: What are this change for ?

Comment (ifaresse) [inline app/build.gradle.kts:1]: What are this change for ?

File: gradle/libs.versions.toml
Comment (ifaresse) [inline gradle/libs.versions.toml:1]: Is it really useful ?
We already have mockito inline, no ?

**Review by ifaresse — APPROVED** [review]
LGTM but answer my comments before merging

File: app/build.gradle.kts
Comment (DanR2003) [reply app/build.gradle.kts:1]: At some point I had to use the mockito.inline dependency when testing

File: gradle/libs.versions.toml
Comment (DanR2003) [reply gradle/libs.versions.toml:1]: no, it looked like we didn't have it, so I had to add it

File: app/build.gradle.kts [hunk 1]
```diff
@@ -190,6 +190,7 @@ dependencies {
     implementation(libs.androidx.runtime.livedata)
     testImplementation(libs.junit)
     testImplementation(libs.androidx.espresso.intents)
+    testImplementation ("androidx.arch.core:core-testing:2.1.0")
```
[the remark this answers is not in the document]
Comment (DanR2003) [reply app/build.gradle.kts]: fixed

File: app/src/androidTest/java/com/swent/suddenbump/ui/calendar/EditMeetingScreenTest.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          friendId = "FPHuqGkCBo7Iinbo5OO9",
+          creatorId = "P7vuP4bbEQB03OSR3QwJ")
+
+  @Before
+  fun setUp() {
+    meetingRepository = mock(MeetingRepository::class.java)
+    navigationActions = mock(NavigationActions::class.java)
+    meetingViewModel = MeetingViewModel(meetingRepository)
+
+    `when`(navigationActions.currentRoute()).thenReturn(Screen.EDIT_MEETING)
```
[the remark this answers is not in the document]
Comment (DanR2003) [reply app/src/androidTest/java/com/swent/suddenbump/ui/calendar/EditMeetingScreenTest.kt]: fixed, I removed the line, as I wasn't using it

File: app/src/main/java/com/swent/suddenbump/ui/calendar/AddMeeting.kt [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import com.swent.suddenbump.model.user.UserViewModel
+import com.swent.suddenbump.ui.navigation.NavigationActions
+import java.text.SimpleDateFormat
+import java.util.*
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AddMeetingScreen(
+    navigationActions: NavigationActions,
+    viewModel: UserViewModel,
```
[the remark this answers is not in the document]
Comment (DanR2003) [reply app/src/main/java/com/swent/suddenbump/ui/calendar/AddMeeting.kt]: fixed, changed variable name to  userViewModel

File: app/src/main/java/com/swent/suddenbump/model/meeting/MeetingRepositoryFirestore.kt
[the remark this answers is not in the document]
Comment (DanR2003) [reply app/src/main/java/com/swent/suddenbump/model/meeting/MeetingRepositoryFirestore.kt:1]: done, removed commented code

**Pull request merged**